Src/Extern/HYPRE/AMReX_HypreABecLap3.cpp

```diff
@@ -0,0 +1,503 @@
+#include <AMReX_HypreABecLap3.H>
+#include <AMReX_HypreABec_F.H>
+#include <cmath>
+
+namespace {
+static int ispow2(int i) {
+  return (i == 1) ? 1 : (((i <= 0) || (i & 1)) ? 0 : ispow2(i / 2));
+}
+}
+
+namespace amrex {
+
+HypreABecLap3::HypreABecLap3(const BoxArray& grids,
+                             const DistributionMapping& dmap,
+                             const Geometry& geom_,
+                             MPI_Comm comm_)
+    : comm(comm_),
+      geom(geom_),
+      verbose(1),
+      A(NULL), b(NULL), x(NULL) {
+
+  const int ncomp = 1;
+  int ngrow = 0;
+  acoefs.define(grids, dmap, ncomp, ngrow);
+  acoefs.setVal(0.0);
+
+  for (int i = 0; i < BL_SPACEDIM; ++i) {
+    BoxArray edge_boxes(grids);
+    edge_boxes.surroundingNodes(i);
+    bcoefs[i].define(edge_boxes, dmap, ncomp, ngrow);
+    bcoefs[i].setVal(0.0);
+  }
+
+  bd.define(grids, dmap, ncomp, geom);
+
+  int num_procs, myid;
+  MPI_Comm_size(comm, &num_procs);
+  MPI_Comm_rank(comm, &myid);
+
+  // initialize with ngrow = 1
+  ngrow = 1;
+
+  // Store the global integer index
+  GbInd.define(grids, dmap, ncomp, ngrow);
+  GbInd.setVal(0);
+
+  // Arrays needed to build the global indices for all procs
+  CellsGIndex.resize(grids.size());
+  numCellsProc.resize(num_procs+1);
+
+  // These arrays store the starting global indices of all the boxes involved
+  StartIndex.resize(grids.size());
+
+  // Initialize all the arrays to 0
+  std::fill(CellsGIndex.begin(), CellsGIndex.end(), 0);
```

Comment:
When the vector is resized, we could do `CellsGIndex.resize(grids.size(), 0);`.  So `std::fill` is unnecessary.